File: Source/Hinshlabs.CommandLineParser.Demo/Commands/Demo2Command/Commands/Demo3CommandLine.vb
Imports Hinshlabs.CommandLineParser
Imports System.Collections.ObjectModel

Public Class Demo3CommandLine
    Inherits CommandLineBase

    Private m_value1 As String
    Private m_value2 As Value2Values = Value2Values.Value1

    <CommandLineSwitch("Value1", "[CommandLineSwitch.Description]"), CommandLineAlias("v1")> _
    Public Property Value1() As String
        Get
            Return Me.m_value1
        End Get
        Set(ByVal value As String)
            Me.m_value1 = value
        End Set
    End Property

    <CommandLineSwitch("Value2", "[CommandLineSwitch.Description]"), CommandLineAlias("v2")> _
    Public Property Value2() As Value2Values
        Get
            Return Me.m_value2
        End Get
        Set(ByVal value As Value2Values)
            Me.m_value2 = value
        End Set
    End Property

    Public Enum Value2Values
        Value1
        Value2
        Value3
    End Enum

End Class

File: Source/Hinshlabs.CommandLineParser.DemoCSharp/Commands/Demo3Command/Demo3CommandLine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hinshlabs.CommandLineParser.DemoCSharp.Commands.Demo3Command
{
    public class Demo3CommandLine : CommandLineBase
    {

        private string m_value1;

        private Value2Values m_value2 = Value2Values.Value1;
        [CommandLineSwitch("Value1", "[CommandLineSwitch.Description]"), CommandLineAlias("v1")]
        public string Value1
        {
            get { return this.m_value1; }
            set { this.m_value1 = value; }
        }

        [CommandLineSwitch("Value2", "[CommandLineSwitch.Description]"), CommandLineAlias("v2")]
        public Value2Values Value2
        {
            get { return this.m_value2; }
            set { this.m_value2 = value; }
        }

        public enum Value2Values
        {
            Value1,
            Value2,
            Value3
        }

    }

}
